pallets/automation-time/src/tests.rs

```diff
@@ -231,19 +236,92 @@ fn schedule_native_transfer_works() {
 	})
 }
 
+#[test]
+fn schedule_xcmp_works() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let call: Vec<u8> = vec![2, 4, 5];
+		get_funds(Sibling::from(PARA_ID).into_account());
+		assert_ok!(AutomationTime::schedule_xcmp_task(
+			cumulus_pallet_xcm::Origin::SiblingParachain(PARA_ID.try_into().unwrap()).into(),
+			vec![50],
+			vec![SCHEDULED_TIME],
+			PARA_ID.try_into().unwrap(),
+			call.clone(),
+			100_000,
+		));
+		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME) {
+			None => {
+				panic!("A task should be scheduled")
+			},
+			Some(task_ids) => match AutomationTime::get_task(task_ids[0]) {
+				None => {
+					panic!("A task should exist if it was scheduled")
+				},
+				Some(task) => {
+					let expected_task = Task::<Test>::create_xcmp_task(
+						Sibling::from(PARA_ID).into_account(),
+						vec![50],
+						vec![SCHEDULED_TIME].try_into().unwrap(),
+						PARA_ID.try_into().unwrap(),
+						call.clone(),
+						100_000,
+					);
+
+					assert_eq!(task, expected_task);
+				},
+			},
+		}
+	})
+}
+
+#[test]
+fn schedule_xcmp_errors_para_id_mismatch() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let para_id_2: u32 = 2001;
+		assert_noop!(
+			AutomationTime::schedule_xcmp_task(
+				cumulus_pallet_xcm::Origin::SiblingParachain(PARA_ID.try_into().unwrap()).into(),
+				vec![50],
+				vec![SCHEDULED_TIME],
+				para_id_2.try_into().unwrap(),
+				vec![3, 4, 5],
+				100_000,
+			),
+			Error::<Test>::ParaIdMismatch,
+		);
+	})
+}
+
+#[test]
+fn schedule_xcmp_errors_bad_origin() {
```

Comment:
should we have a variety of cases here for Bad Origin? i.e. i think there are a bunch of ways to trigger it, right?
1. if the sovereign account does not match the Para ID
2. If the OriginType is SovereignAccount rather than Native
3. May be some other way to trigger that I don't know?

Reply:
1. this is tested here `schedule_xcmp_errors_para_id_mismatch` and returns the error `ParaIdMismatch`
2. I fixed this test and am passing a `SovereignAccount` and that returns `BadOrigin`